docs/design/proposals/direct-blk-device-assignment.md

```diff
@@ -0,0 +1,264 @@
+# Background
+Today, there exists a few gaps between Container Storage Interface (CSI) and virtual-machine(VM) based runtimes like Kata Containers 
+that prevent them from working together smoothly.
+
+First, it’s cumbersome to use a persistent volume (PV) with Kata Containers. Today, for a PV with Filesystem volume mode, `virtiofs`
+is the only way to surface it insider a Kata guest VM. But often mounting the filesystem (FS) within the guest environment is 
+desired due to performance benefits, availability of native FS features and security benefits over the `virtiofs` mechanism.
+
+Second, it’s difficult if not impossible to resize a PV online with Kata Containers. While a PV can be expanded on the host OS, 
+the updated metadata needs to be propagated to the guest OS in order for the application container to use the expanded volume. 
+Currently, there does not exist a way to propagate the PV metadata from the host OS to the guest OS without restarting the sandbox.
+
+Our use cases at Databricks need two features in the Kata Containers: support PV Filesystem volume mode with native FS types and expand a PV online.
+
+# Proposed Solution
+
+Because of the OS boundary, these features cannot be implemented in the CSI driver plugin running on the host OS 
+as is normally done in the runc container. Instead, they can be done by the kata agent inside the guest OS, 
+but it requires the CSI driver to pass the relevant information to the Kata container runtime. 
+An ideal long term solution would be to have the kubelet coordinating the communication between the CSI driver and 
+container runtime, as described in [KEP-2857](https://github.com/kubernetes/enhancements/pull/2893/files). 
+However, as the KEP is still under review, we would like to propose a short/medium term solution to unblock our use case.
+
+The proposed solution is built on top of a previous [proposal](https://github.com/egernst/kata-containers/blob/da-proposal/docs/design/direct-assign-volume.md) 
+described by Eric Ernst. The previous proposal has two gaps:
+
+1. Writing a `csiPlugin.json` file to the volume root path introduces a security risk. A malicious user can gain unauthorized 
+access to a block device by writing their own `csiPlugin.json` to the above location through an ephemeral CSI plugin.  
+
+2. The proposal doesn’t describe how to establish a mapping between a volume and a kata sandbox, which is needed for 
+implementing CSI volume resize and volume stat collection APIs.
+
+This document particularly focuses on how to address these two gaps.
+
+## Assumptions
+The proposal assumes that a block device volume will only be used by one Pod on a node at a time, which we believe 
+is the more common pattern in the Kata container. It’s rare and unsafe to have the same block device attached to more than 
+one Kata Pods. In the context of Kubernetes, the PersistentVolumeClaim (PVC) needs to have accessMode as `ReadWriteOncePod`. 
```

Comment:
Can we also call out under Assumptions that Kubernetes pod specs that specify [1] fsGroup, [2] fsGroupChangePolicy and [3] subPaths as part of their container mounts won't be supported in the context of this proposal?